table/format.cc

```diff
@@ -352,6 +358,8 @@ Status UncompressBlockContentsForCompressionType(
           BlockContents(std::move(ubuf), decompress_size, true, kNoCompression);
       break;
     case kXpressCompression:
+      // XPRESS allocates memory internally, thus no support for custom
+      // allocator.
```

Comment:
To handle compression lib that don't allow external allocation, how about adding a `CacheAllocationPtr WrapRawAllocation(void*)` method to allocator so it can optionally:
* return the pointer as is but bind with a different deleter, or 
* take the perf penalty and copy the memory into it is own memory buffer